VCToken/VCToken/Jwk.swift

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import Foundation
+
+enum JwkError: Error {
+    case invalidKeyValue
+}
+
+/// Runtime container for JSON Web Keys, in which key material is Base64URLEncoded
+public struct Jwk: Codable, Equatable {
+    
+    public let keyType: String
+    public let keyId: String?
+    public let key: Data?
+    public let curve: String?
+    public let use: String?
+    public let x: Data?
+    public let y: Data?
+    public let d: Data?
+    
+    enum CodingKeys: String, CodingKey {
+        case keyType = "kty"
+        case keyId = "kid"
+        case key = "k"
+        case curve = "crv"
+        case use, x, y, d
+    }
+
+    public init(keyType: String,
+                keyId: String? = nil,
+                key: Data? = nil,
+                curve: String? = nil,
+                use: String? = nil,
+                x: Data? = nil,
+                y: Data? = nil,
+                d: Data? = nil) {
+        self.keyType = keyType
+        self.keyId = keyId
+        self.key = key
+        self.curve = curve
+        self.use = use
+        self.x = x
+        self.y = y
+        self.d = d
+    }
+    
+    public init(from decoder: Decoder) throws {
+
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        func parseKeyIfPresent(_ key: KeyedDecodingContainer<Jwk.CodingKeys>.Key) throws -> Data? {
```

Comment:
what does this do?

Reply:
It decodes the value for `key`, if present in the JSON, as Base64URL-encoded binary data, which isn't the default for binary data under `JSONEncoder` I discovered